networktable.c

```diff
@@ -41,17 +41,101 @@ bool create_network_table(void)
 bool create_network_addresses_table(void)
 {
 	bool ret;
+
+	dbquery("BEGIN TRANSACTION");
+
 	// Create network_addresses table in the database
 	ret = dbquery("CREATE TABLE network_addresses ( network_id INTEGER NOT NULL, "\
 	                                               "ip TEXT NOT NULL, "\
-	                                               "lastSeen INTEGER NOT NULL, "\
+	                                               "lastSeen INTEGER NOT NULL DEFAULT (cast(strftime('%%s', 'now') as int)), "\
 	                                               "UNIQUE(network_id,ip), "\
 	                                               "FOREIGN KEY(network_id) REFERENCES network(id));");
-	if(!ret){ dbclose(); return false; }
+	if(!ret)
+	{
+		logg("create_network_addresses_table(): CREATE TABLE network_addresses failed!");
+		dbclose();
+		return false;
+	}
+
+	// Create a network_addresses row for each entry in the network table
+	ret = dbquery("INSERT INTO network_addresses (network_id,ip) SELECT id,ip FROM network;");
+	if(!ret)
+	{
+		logg("create_network_addresses_table(): INSERT INTO network_addresses failed!");
+		dbclose();
+		return false;
+	}
+
+	// Remove IP column from network table.
+	// As ALTER TABLE is severely limit, we have to do the column deletion manually.
+	// Step 1: We create a new table without the ip column
+	//         We add the UNIQUE constraint on (ip,hwaddr) already here
+	ret = dbquery("CREATE TABLE network_bck ( id INTEGER PRIMARY KEY NOT NULL, " \
+	                                         "hwaddr TEXT NOT NULL, " \
+	                                         "interface TEXT NOT NULL, " \
+	                                         "name TEXT, " \
+	                                         "firstSeen INTEGER NOT NULL, " \
+	                                         "lastQuery INTEGER NOT NULL, " \
+	                                         "numQueries INTEGER NOT NULL, " \
+	                                         "macVendor TEXT, " \
+	                                         "UNIQUE(id,hwaddr));");
```

Comment:
If we just have a unique constraint on `hwaddr`, that would be even better than on on `(id, hwaddr)`.

Reply:
Why do you think so? I don't think it would make much of a difference, but we can surely change it.

Reply:
It fits the data better. The important thing is that the MAC address is unique, not necessarily that it is unique with respect to some arbitrary ID. With the previous code, you could have duplicate MAC addresses as long as they have a different ID.